components/mlmd/Get_artifact_types/component.py

```diff
@@ -0,0 +1,32 @@
+from kfp.components import create_component_from_func
+
+def get_artifact_types_from_mlmd() -> list:
+    """Gets artifact types from MLMD
+
+    Annotations:
+        author: Alexey Volkov <alexey.volkov@ark-kun.com>
+    """
+    import os
+    from google.protobuf.json_format import MessageToDict
+    from ml_metadata.proto import metadata_store_pb2
+    from ml_metadata.metadata_store import metadata_store
+    metadata_service_host = os.environ.get('METADATA_GRPC_SERVICE_SERVICE_HOST', 'metadata-grpc-service')
```

Comment:
Maybe default to `metadata-grpc-service.kubeflow.svc.cluster.local` for the cases in which the pods run in different namespaces?
If the service is in the same namespace, the env will be there no matter what.
If some namespace configuration is "special" and unpredicted (e.g., KFP deployed in a custom namespace), then users should configure the env accordingly.

Reply:
I've seen KFP installed in the `kubeflow` and `default` namespaces.
Are you sure your suggested change will always work?
What are the scenarios when `'metadata-grpc-service'` does not work?

Reply:
@Ark-kun 
>Are you sure your suggested change will always work?

Well, it cannot always work if there is no "discovery" of where the metadata service is deployed.

>What are the scenarios when `'metadata-grpc-service'` does not work?

If the metadata service is not deployed in the same namespace as the pod (the KFP run step in this case), then this default value will fail.
A plain `<SERVICE_NAME>` value leads to the DNS `<SERVICE_NAME>.<MY_NAMESPACE>.svc.cluster.local`.

Quoting from https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#namespaces-of-services:
>For example, consider a pod in a `test` namespace. A data service is in the `prod` namespace.
A query for `data` returns no results, because it uses the pod's `test` namespace.
A query for `data.prod` returns the intended result, because it specifies the namespace.

So, this default value only works if the service is deployed in the same namespace with the KFP run. **But** if that's the case, then the kubelet has set the `METADATA_GRPC_SERVICE_SERVICE_HOST` env var (https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables).
As a result, this default value is not necessary.

That's the reason why we need to be setting a default value that does specify a namespace for the service: because we expect [and **know**] that the service lives in a different namespace from the pod when we get to actually need this value.
Choosing the `kubeflow` namespace sounds as the most sane default value, since most users deploy KF in the `kubeflow` namespace using https://github.com/kubeflow/manifests.